substrate/frame/revive/src/evm/runtime.rs

```diff
@@ -0,0 +1,636 @@
+	fn new(function: Call, signed_data: Option<Self::SignaturePayload>) -> Option<Self> {
+		Some(if let Some((address, signature, extra)) = signed_data {
+			Self(generic::UncheckedExtrinsic::new_signed(function, address, signature, extra))
+		} else {
+			Self(generic::UncheckedExtrinsic::new_unsigned(function))
+		})
+	}
+}
+
+impl<Address, Call, Signature, E: EthExtra> ExtrinsicMetadata
+	for UncheckedExtrinsic<Address, Call, Signature, E>
+{
+	const VERSION: u8 = generic::UncheckedExtrinsic::<Address, Call, Signature, E::Extra>::VERSION;
+	type SignedExtensions = E::Extra;
+}
+
+impl<Address: TypeInfo, Call: TypeInfo, Signature: TypeInfo, E: EthExtra> ExtrinsicCall
+	for UncheckedExtrinsic<Address, Call, Signature, E>
+{
+	fn call(&self) -> &Self::Call {
+		self.0.call()
+	}
+}
+
+use sp_runtime::traits::MaybeDisplay;
+type OnChargeTransactionBalanceOf<T> = <<T as pallet_transaction_payment::Config>::OnChargeTransaction as OnChargeTransaction<T>>::Balance;
+
+impl<LookupSource, Call, Signature, E, Lookup> Checkable<Lookup>
+	for UncheckedExtrinsic<LookupSource, Call, Signature, E>
+where
+	E: EthExtra,
+	Self: Encode,
+	<E::Config as frame_system::Config>::Nonce: TryFrom<U256>,
+	<E::Config as frame_system::Config>::RuntimeCall: Dispatchable<Info = DispatchInfo>,
+	OnChargeTransactionBalanceOf<E::Config>: Into<BalanceOf<E::Config>>,
+	BalanceOf<E::Config>: Into<U256> + TryFrom<U256>,
+	MomentOf<E::Config>: Into<U256>,
+	Call: From<crate::Call<E::Config>> + TryInto<crate::Call<E::Config>>,
+
+	// required by Checkable for `generic::UncheckedExtrinsic`
+	LookupSource: Member + MaybeDisplay,
+	Call: Encode + Member,
+	Signature: Member + traits::Verify,
+	<Signature as traits::Verify>::Signer: IdentifyAccount<AccountId = AccountIdOf<E::Config>>,
+	E::Extra: SignedExtension<AccountId = AccountIdOf<E::Config>>,
+	Lookup: traits::Lookup<Source = LookupSource, Target = AccountIdOf<E::Config>>,
+{
+	type Checked = CheckedExtrinsic<AccountIdOf<E::Config>, Call, E::Extra>;
+
+	fn check(self, lookup: &Lookup) -> Result<Self::Checked, TransactionValidityError> {
+		if self.0.signature.is_none() {
+			if let Ok(call) = self.0.function.clone().try_into() {
+				if let crate::Call::eth_transact { payload, gas_limit, storage_deposit_limit } =
+					call
+				{
+					let checked = E::try_into_checked_extrinsic(
+						payload,
+						gas_limit,
+						storage_deposit_limit,
+						self.encoded_size(),
+					)?;
+					return Ok(checked)
+				};
+			}
+		}
+		self.0.check(lookup)
+	}
+
+	#[cfg(feature = "try-runtime")]
+	fn unchecked_into_checked_i_know_what_i_am_doing(
+		self,
+		lookup: &Lookup,
+	) -> Result<Self::Checked, TransactionValidityError> {
+		self.0.unchecked_into_checked_i_know_what_i_am_doing(lookup)
+	}
+}
+
+impl<Address, Call, Signature, E> GetDispatchInfo
+	for UncheckedExtrinsic<Address, Call, Signature, E>
+where
+	Call: GetDispatchInfo,
+	E: EthExtra,
+{
+	fn get_dispatch_info(&self) -> DispatchInfo {
+		self.0.get_dispatch_info()
+	}
+}
```

Comment:
no we want it like this so that when you ask the weight of eth_transact it returns MAX
it's only used by runtime api api like  tx_payment to get query_fee_details as far as I can tell

Reply:
Isn't it also used during block building to quickly estimate if the tx will fit? Seems dangerous to me to return the wrong thing here.

Reply:
It's not the executive do it like this
https://github.com/paritytech/polkadot-sdk/blob/483b6385eb631d412e0124f6169256768ac5a9cc/substrate/frame/executive/src/lib.rs?plain=1#L785-L787

it's really just these 2 runtime APIs query_fee_details.

> Seems dangerous to me to return the wrong thing here.

Well we return the truth :), would be pretty bad that the logic to get the underlying call and it's dispatch info was executed twice.

Reply:
Its also called in `validate_transaction` but only on checked transaction. So this should be fine:
https://github.com/paritytech/polkadot-sdk/blob/483b6385eb631d412e0124f6169256768ac5a9cc/substrate/frame/executive/src/lib.rs?plain=1#L868-L874

I am wondering if the unchecked transaction should even imlement `GetDispatchInfo`. It seems like a good gun. Those `query_fee_details` runtime APIs will return the wrong values.